### PR TITLE
UCT/IB: Fix upstream RoCE DC

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/mlx5/ib_mlx5.inl [hunk 1]
```diff
@@ -217,7 +217,7 @@ uct_ib_mlx5_set_dgram_seg(struct mlx5_wqe_datagram_seg *seg,
 
     if (grh_av) {
         ucs_assert(av->dqp_dct & UCT_IB_MLX5_EXTENDED_UD_AV);
-#if HAVE_STRUCT_MLX5_GRH_AV_RMAC
+#if HAVE_STRUCT_MLX5_GRH_AV_RMAC || !HAVE_STRUCT_MLX5_WQE_AV_BASE
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5.inl]: why it's a problem if HAVE_STRUCT_MLX5_WQE_AV_BASE is defined?

Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/ib_mlx5.inl]: struct mlx5_wqe_av defined differently in mlx5dv.h and mlx5_hw.h
we have compat layer to work with it transparently - see https://github.com/openucx/ucx/blob/master/src/uct/ib/mlx5/ib_mlx5.h#L72
HAVE_STRUCT_MLX5_WQE_AV_BASE means mlx5_hw.h case.

Comment (yosefe) [reply src/uct/ib/mlx5/ib_mlx5.inl]: so in mlx5_hw.h we don't need to copy rmac? who sets it then?

Comment (yosefe) [reply src/uct/ib/mlx5/ib_mlx5.inl]: maybe in ib_mlx5.h, define HAVE_STRUCT_MLX5_GRH_AV_RMAC to 1 in mlx5_hw.h case?

Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/ib_mlx5.inl]: mlx5_hw.h case (HAVE_STRUCT_MLX5_WQE_AV_BASE defined) - in this case we need to check HAVE_STRUCT_MLX5_GRH_AV_RMAC, it was added to ofed at some point.
mlx5dv.h case (HAVE_STRUCT_MLX5_GRH_AV_RMAC not defined) - we define struct, not need to check

Comment (yosefe) [reply src/uct/ib/mlx5/ib_mlx5.inl]: i think force-defining HAVE_STRUCT_MLX5_GRH_AV_RMAC after we define the struct will be more self-explanatory than !HAVE_STRUCT_MLX5_WQE_AV_BASE

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11831/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>UNKNOWN</b> on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 18-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/2e53743fd93441752bb11c2711ef930222582e27/_main) | :question: ABORTED
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/3488541961a8fe0f38bebf9f825ef124d22df24c/hpc-test-node-gpu_W0) | :question: ABORTED
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/494feb2b4056893415f0c18c738ec9aacb380a86/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/f0087c53d97d200fbfcbda7ea474c5c86ff17686/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/b1e5447cb40278d2b540ae45208672bda3eac667/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/58d5f1d7529e2e3d36fe46d184a150f958a372f1/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/3a24b6d4a7267ff2b8fc0182266ffa91ec718f01/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/86e94b7f3a1c27670a210b0a5cd99aa376a96774/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/3860f3e330928a1bd8ff64d2c471b9486623f0b2/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/2ad8a011284f89476b30b4c406981593f1a1de7a/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/3fa4e1fad8524d97fb7fefb8c8834f7c8195f473/hpc-arm-cavium-jenkins_W0) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/d4e6deb1a3f44a515e424f22f0dea0fd51e8227c/hpc-arm-cavium-jenkins_W1) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/ca247451350efba76bd566369d7f6ee7bddf397e/hpc-arm-cavium-jenkins_W2) | :question: UNKNOWN
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/5226743fb1bd9334247bd59e3cc8913637a5b0bf/hpc-arm-cavium-jenkins_W3) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/a7a73da9b5696badd5cbd6447f4231be3ad0dfbd/hpc-arm-hwi-jenkins_W0) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/cfb13d0abd97cfeefb3d492c8855d1ac0bb2fb1f/hpc-arm-hwi-jenkins_W1) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/e4f6169b6bf267a3ef5680849ab9e0d8f7d8edc6/hpc-arm-hwi-jenkins_W2) | :question: UNKNOWN
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/c77c7f4cdeeca25d86b01a1cdc6c308db4c0f2a0/hpc-arm-hwi-jenkins_W3) | :question: UNKNOWN
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/75458170b2aef87d83e2bc2b44d9639f4235af91/hpc-test-node-gpu_W1) | :question: UNKNOWN
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/9f590ed45c7ebee831def6602a4bd713bd520766/hpc-test-node-gpu_W2) | :question: UNKNOWN
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/6ac839c012584d859ee6c0ab5c30266fc5f3a901/hpc-test-node-gpu_W3) | :question: UNKNOWN
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/b56c8f5857041024cd941f92488b0e6957c285d3/hpc-test-node-legacy_W0) | :question: UNKNOWN
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/a8ee44a6ec0103b500478714702d4a43e091f204/hpc-test-node-legacy_W1) | :question: UNKNOWN
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/30de315a58973262e12ca7beec9fc21ee00ba677/hpc-test-node-legacy_W2) | :question: UNKNOWN
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/d351e9434a9ceb44f04ac4165c9f3a43ebede235/hpc-test-node-legacy_W3) | :question: UNKNOWN
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/7a821a19e8f0be7da063be62c2475b67501d8a66/hpc-test-node-new_W0) | :question: UNKNOWN
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/89c927309bc2e761603ef8893028c8edef0a1802/hpc-test-node-new_W1) | :question: UNKNOWN
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/7e9b46cefd1e5f92dd3a8b69a28e9bc788ccbad7/hpc-test-node-new_W2) | :question: UNKNOWN
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/be5675b6098478eb88970c4b7c76b978/raw/733daa5de1f30c911f42f997e3ce94b15abb7920/hpc-test-node-new_W3) | :question: UNKNOWN

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11835/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 18-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/ec6df8346be7aaed395dbfb915f9cab9255b294e/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/8e52c6df40ffc8322269248012e6a00b415bd4b7/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/dd43dbedae7d11dfe1212d8990cc0ec8752318b3/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/654e907bccb45c5fcf79ea633af93beca2beb1d6/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/710ba3e8258c0caf7da2247d36d21f4c11f08152/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/fc3fe829f644bf403f56e8f8d0d3fecd43206040/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/936e34475be4cb15092f887766cb36118fa57d20/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/94094f87bd13f38843bdba7836e66225a6e694ea/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/7b002084e862c6a2e5633ca250d25a0c90fe969f/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/be306863b0dcda6dd8f4cbb5766bc3ad381545c1/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/d9e11336ed316106e92c14983d9a7cca6a6e1b29/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/6578509fe3a7232e12098b03c25b1200ddead162/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/d127d967626a70c689acb5787adfbae1d40013e2/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/2b1d691ac9cfdf97770038a7a6b1c243c77f4431/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/6a8ce20c896fa0c132a414825e163cf358469233/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/f270fcb9b81fed8b5d10c53d5afdd7bbc5de5c3e/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/5a32e6547ae1f1f0ae22e08dd0418537e1e71d5e/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/40146fed35648d816d95b6931eafd35e79247b40/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/c7b13d791793de5e38840581b8dffc495dcb6f5d/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/6cb777831ad560a0b0e185c670af60737a919b90/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/05a48c49b6ec37b146cc7e1d5cf66b93d5d7279f/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/976d16599ff5ac3ece62414b88691b52e1662dec/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/015db451309a37a923a5e8af45ab94a7fde4d675/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/fca692423d654b255c3ca6f2cf631e404d541211/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/55a90ca018cbc225129a7824178c660bd373739e/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/03db6d7cedc23e71e4ddd34821755650f327dfb6/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/74b41c16873dd92e528bdac7239721533c2d0978/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/bbeeb9f8018b90ccde6c66437eee493df8809895/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/0b68720fa91eaf7a5e41ca7fe88ad621/raw/a74c26b9349b1f5ce663177cf28982ed7b99a39f/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
boo31 failure
bot:pipe:retest

**Pull request merged**